app/src/test/kotlin/co/netguru/baby/monitor/client/feature/voiceAnalysis/VoiceAnalysisUseCaseTest.kt

```diff
@@ -0,0 +1,118 @@
+package co.netguru.baby.monitor.client.feature.voiceAnalysis
+
+import co.netguru.baby.monitor.client.common.ISchedulersProvider
+import co.netguru.baby.monitor.client.common.Randomiser
+import co.netguru.baby.monitor.client.data.DataRepository
+import co.netguru.baby.monitor.client.feature.communication.websocket.Message
+import co.netguru.baby.monitor.client.feature.communication.websocket.MessageController
+import com.nhaarman.mockitokotlin2.*
+import io.reactivex.Completable
+import io.reactivex.Observable
+import io.reactivex.schedulers.Schedulers
+import io.reactivex.schedulers.TestScheduler
+import org.junit.Test
+import java.util.concurrent.TimeUnit
+
+class VoiceAnalysisUseCaseTest {
+
+    private val dataRepository = mock<DataRepository> {
+        on { updateVoiceAnalysisOption(any()) }.doReturn(Completable.complete())
+    }
+    private val randomDigitsList = listOf(1, 2, 3, 4)
+    private val confirmationId = randomDigitsList.joinToString("")
+    private val timerTestScheduler = TestScheduler()
+    private val schedulersProvider = mock<ISchedulersProvider> {
+        on { io() } doReturn timerTestScheduler
+        on { mainThread() } doReturn Schedulers.trampoline()
+        on { computation() } doReturn Schedulers.trampoline()
+    }
+    private val messageController = mock<MessageController> {
+        on { receivedMessages() }.doReturn(Observable.just(Message(confirmationId = confirmationId)))
+    }
+    private val randomiser = mock<Randomiser> {
+        on { getRandomDigits(any()) }.doReturn(randomDigitsList)
+    }
+
+    private val voiceAnalysisUseCase =
+        VoiceAnalysisUseCase(dataRepository, schedulersProvider, randomiser)
+
+    @Test
+    fun `should send noiseDetection message to baby device`() {
+        voiceAnalysisUseCase.chooseVoiceAnalysisOption(
+            messageController,
+            VoiceAnalysisOption.NoiseDetection
+        ).subscribe()
+
+        verify(messageController).sendMessage(argThat { voiceAnalysisOption == VoiceAnalysisOption.NoiseDetection.name })
+    }
+
+    @Test
+    fun `should send machineLearning message to baby device`() {
+        voiceAnalysisUseCase.chooseVoiceAnalysisOption(
+            messageController,
+            VoiceAnalysisOption.MachineLearning
+        ).subscribe()
+
+        verify(messageController).sendMessage(argThat { voiceAnalysisOption == VoiceAnalysisOption.MachineLearning.name })
+    }
+
+    @Test
+    fun `should get successful response with the same confirmationId`() {
+        voiceAnalysisUseCase.chooseVoiceAnalysisOption(
+            messageController,
+            VoiceAnalysisOption.MachineLearning
+        ).test()
+            .assertValue(true)
+
+        verify(messageController).sendMessage(argThat { voiceAnalysisOption == VoiceAnalysisOption.MachineLearning.name })
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/netguru/baby-monitor-client-android/blob/89ff9ffb603a83b79e177dd00519720c6a8d52ac/app/src/test/kotlin/co/netguru/baby/monitor/client/feature/voiceAnalysis/VoiceAnalysisUseCaseTest.kt#L67"></span>Line detected that is longer than the defined maximum line length in the code style.</td>
    </tr>
  </tbody>
</table>